src/feature/consulenze/components/ReservationModal.tsx

```diff
@@ -18,12 +19,23 @@ export const ReservationModal: React.FC<RouteComponentProps> = () => {
     navigate("/consulenze/");
   }, [navigate]);
 
+  const handleBack = React.useCallback(() => {
+    if (step === "welcome") handleClose();
+    prevStep();
+  }, [step, handleClose, prevStep]);
```

Comment:
Usa la funzione `onClose` del modalContext  direttamente all'interno del WelcomeModal e passa ad entrambi quella funzione.

Reply:
In che senso scusa?

Reply:
`useModalContext` ritorna la funzione di onClose che passiamo al Modal. 
Dentro il tuo `WelcomeModal` richiami l'hook e sia al bottone per andare indietro e a quello per chiudere passi quella funzione. Senza il bisogno di definire questa funzione.

Reply:
Ok, tra l'altro pensavo che potremmo aggiungere al modalContext anche i metodi onBack ed onContinue, e fare lo stesso gioco